AdyenDropIn/Utilities/ComponentManager.swift

```diff
@@ -102,6 +102,21 @@ internal final class ComponentManager {
 
     internal lazy var paidComponents = paymentMethods.paid.compactMap(component(for:))
     
+    /// Returns the only regular component that is not an instant payment,
+    /// when no other payment method exists.
+    internal var singleRegularComponent: (PaymentComponent & PresentableComponent)? {
+        guard storedComponents.isEmpty,
+              paidComponents.isEmpty,
+              regularComponents.count == 1,
+              let regularComponent = regularComponents.first else { return nil }
```

Comment:
you can fail early here, like this:

```Swift
guard storedComponents.isEmpty,
               paidComponents.isEmpty,
               regularComponents.count == 1,
               let regularComponent = regularComponents.first as? (PaymentComponent & PresentableComponent) else {
 return nil
}
return regularComponent
```

no need to check `InstantPaymentComponent ` since its not a `PresentableComponent ` :-)

Reply:
Ahh ok, good to know